packages/snap-client/src/Client/apis/Recommend.ts

```diff
@@ -128,7 +146,82 @@ export class RecommendAPI extends API {
 		return deferred.promise;
 	}
 
-	async getRecommendations(queryParameters: RecommendRequestModel): Promise<RecommendResponseModel> {
+	transformFilters(filters: filter[], type: 'POST' | 'GET') {
+		if (type == 'POST') {
+			const filterArray: filtersObj[] = [];
+			filters.map((filter) => {
+				if (filter.type == 'value') {
+					//check if filterArray contains a filter for this value already
+					const i = filterArray.findIndex((_filter) => _filter.field == filter.field);
+					if (i > -1) {
+						//if so just push another value to the already existing obj
+						(filterArray[i].values as string[]).push(filter.value);
+					} else {
+						//else create a new one
+						const val = {
+							field: filter.field,
+							type: '=' as const,
+							values: [filter.value],
+						};
+
+						filterArray.push(val);
+					}
+				} else if (filter.type == 'range') {
+					//low
+					if (typeof filter.value.low == 'number') {
+						const low = {
+							field: filter.field,
+							type: '>=' as const,
+							values: [filter.value.low as number],
+						};
+						filterArray.push(low);
+					}
+					//high
+					if (typeof filter.value.high == 'number') {
+						const high = {
+							field: filter.field,
+							type: '<=' as const,
+							values: [filter.value.high as number],
+						};
+						filterArray.push(high);
+					}
+				}
+			});
+
+			return filterArray;
+		} else if (type == 'GET') {
+			const filterArray: {
+				[filter: `filter.${string}`]: string | number | string[] | number[];
+			} = {};
+
+			filters.map((filter) => {
+				if (filter.type == 'value') {
+					//check if filterArray contains a filter for this value already
+					if (filterArray[`filter.${filter.field}`]) {
+						// is the existing filter an array already? or just a single value
+						if (typeof filterArray[`filter.${filter.field}`] == 'string') {
+							filterArray[`filter.${filter.field}`] = [filterArray[`filter.${filter.field}`] as string, filter.value];
```

Comment:
This breaks once the existing value is an array. Better to always have it an array and just push to it.

<img width="493" alt="image" src="https://github.com/searchspring/snap/assets/5060400/044c8eae-cfb0-4d91-86af-c18d0ff733d5">